### PR TITLE
Add option to enable GNU Install Dirs

### PR DESCRIPTION
**Pull request by jpgr87**
The way issue 15 was solved made it very difficult to install console_bridge system-wide on a distribution that does not use /usr/lib as the library installation path. This patch is a better way to solve the issue 15: rather than just nuking the GNUInstallDirs logic from orbit, it adds a configuration switch to optionally enable the functionality, leaving it disabled by default so ROS doesn't break.

### CODE REVIEW

**Comment by scpeters** [comment]
I think this was resolved by #23

**Comment by dirk-thomas** [comment]
The same change has been reverted a while ago since it did not work on the ROS build farm: https://github.com/ros/console_bridge/commit/b863035a68d9a4ea194dfd2872bbd32354f64b15

I am not sure if something is different now though.

**Comment by jpgr87** [comment]
As I said in the original description, this is a better way to solve #15 (the buildfarm breakage) by adding an off-by-default option to enable paths from GNUInstallDirs. 

The motivation behind this PR is the fact that when libraries are installed system-wide, distros don't always put all of their libraries in `${prefix}/lib/`.  GNUInstallDirs provides a set of heuristics to choose the correct library installation directory for a given environment.  On the other hand, packages installing to non-standard prefixes like `/opt/ros` don't necessarily want or need to follow the distro library directory conventions.  The patch is supposed to preserve the hard-coded `${prefix}/lib/` destination by default, and provide an option to allow distro maintainers to use paths derived from GNUInstallDirs instead.

#22 and #23 don't fix this issue, as they don't address the fact that the library installation directory is hard-coded to `${prefix}/lib/`

**Comment by scpeters** [comment]
Can you resolve conflicts on this branch?

**Comment by jpgr87** [comment]
Should be resolved now.

**Comment by j-rivero** [comment]
AFAIK, the support for `GNUInstallsDirs` [was reverted](https://github.com/ros/console_bridge/commit/b863035a68d9a4ea194dfd2872bbd32354f64b15) some time ago to keep compatibility with catkin before [we merged the support for it](https://github.com/ros/catkin/pull/624).

So two things: 
 * I would say that this can be now merged and work within ROS 
 * I don't see any reason to implement it via optional variable.

File: console_bridge-config.cmake.in [hunk 1]
```diff
@@ -8,7 +8,7 @@ set(@PKG_NAME@_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include")
 foreach(lib @PKG_LIBRARIES@)
```
Comment (j-rivero) [inline console_bridge-config.cmake.in]: in the line 6 of this file, we can change:

```
-set(@PKG_NAME@_INCLUDE_DIRS "@CMAKE_INSTALL_PREFIX@/include")
+set(@PKG_NAME@_INCLUDE_DIRS @CMAKE_INSTALL_FULL_INCLUDEDIR@)
```

File: console_bridge.pc.in [hunk 2]
```diff
@@ -1,8 +1,8 @@
 # This file was generated by CMake for @PROJECT_NAME@
 prefix=@CMAKE_INSTALL_PREFIX@
 exec_prefix=${prefix}
-libdir=${prefix}/lib
-includedir=${prefix}/include
+libdir=${prefix}/@CMAKE_INSTALL_LIBDIR@
```
Comment (j-rivero) [inline console_bridge.pc.in]: I'm patch this in https://github.com/ros/console_bridge/pull/29/files

Comment (jpgr87) [inline console_bridge.pc.in]: Should I remove this part, or incorporate the changes from #29?

File: console_bridge-config.cmake.in (hunk 1 above)
Comment (jpgr87) [inline console_bridge-config.cmake.in]: True, I will fix this.

File: console_bridge.pc.in (hunk 2 above)
Comment (j-rivero) [inline console_bridge.pc.in]: > Should I remove this part, or incorporate the changes from #29?

It only makes sense when using GNUInstallDirs, so probably the right way would to incorporate changes here and close pull request #29.

Comment (scpeters) [inline console_bridge.pc.in]: oops, I merged #29 already

sorry

Comment (jpgr87) [inline console_bridge.pc.in]: No problem, I'll merge it back into this PR

File: CMakeLists.txt [hunk 3]
```diff
@@ -82,7 +82,7 @@ if (NOT MSVC)
   set(pkg_conf_file "console_bridge.pc")
   configure_file("${pkg_conf_file}.in" "${CMAKE_BINARY_DIR}/${pkg_conf_file}" @ONLY)
   install(FILES "${CMAKE_BINARY_DIR}/${pkg_conf_file}"
-    DESTINATION lib/pkgconfig/ COMPONENT pkgconfig)
+    DESTINATION ${CMAKE_INSTALL_LIBDIR}/pkgconfig/ COMPONENT pkgconfig)
```
Comment (j-rivero) [inline CMakeLists.txt]: We probably need to use `CMAKE_INSTALL_LIBDIR` with the cmake module, line 72

Comment (jpgr87) [inline CMakeLists.txt]: Yeah, I would agree with that.  Pushing now.

**Comment by j-rivero** [comment]
Thanks Rich for all the fixes. It is fine for me to merge. +1

File: CMakeLists.txt [hunk 4]
```diff
@@ -45,11 +45,11 @@ set_target_properties(${PROJECT_NAME} PROPERTIES SOVERSION
...
 
 install(TARGETS ${PROJECT_NAME}
-  DESTINATION lib/)
+  DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
 install(DIRECTORY include/
-  DESTINATION include
-  FILES_MATCHING PATTERN "*.h")
+  DESTINATION ${CMAKE_INSTALL_INCLUDEDIR} 
+  FILES_MATCHING PATTERN "*.h")
```
Comment (scpeters) [inline CMakeLists.txt]: Could you remove the extra whitespace at the end of these lines?

**Comment by scpeters** [comment]
+1, merging, thanks @jpgr87 !

**Pull request merged**